spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -550,25 +549,23 @@ private Object doGet(final Message<?> requestMessage) {
 
 				@Override
 				public File doInSession(Session<F> session) throws IOException {
-					return AbstractRemoteFileOutboundGateway.this.get(requestMessage, session, remoteDir, remoteFilePath,
-							remoteFilename, true);
+					return AbstractRemoteFileOutboundGateway.this.get(requestMessage, session, remoteDir,
+							remoteFilePath, remoteFilename, true);
 
 				}
 			});
 		}
-		AbstractIntegrationMessageBuilder<Object> builder = this.getMessageBuilderFactory().withPayload(payload)
-			.setHeader(FileHeaders.REMOTE_DIRECTORY, remoteDir)
-			.setHeader(FileHeaders.REMOTE_FILE, remoteFilename);
-		if (session != null) {
-			builder.setHeader(FileHeaders.REMOTE_SESSION, session);
-		}
-		return builder.build();
+		return getMessageBuilderFactory().withPayload(payload)
+				.setHeader(FileHeaders.REMOTE_DIRECTORY, remoteDir)
+				.setHeader(FileHeaders.REMOTE_FILE, remoteFilename)
+				.setHeader(FileHeaders.REMOTE_SESSION, session)
```

Comment:
I see you're relying on `null` not being added here - I guess ignore my comment above.